camligorkem/assignment3/WordSearch.java

```diff
@@ -0,0 +1,123 @@
+import java.util.HashSet;
+import java.util.Optional;
+
+public class WordSearch {
+
+    /**
+     * Given a grid and a dictionary it finds the words (isWord=true) that exists in the dictionary
+     * and returns hashset of strings that are words
+     * @param dictionary is made up a list of entries which are either words or prefixes
+     * @param grid the board with the allowed letters
+     * @return return a hashset of strings that are found to be words (isWord = true)
+      */
+    public static Optional<HashSet<String>> findWords(Dictionary dictionary, char[][] grid ){
```

Comment:
I think Optional here is unnecessary. Returning an empty set should be completely fine. Or do you see any specific benefit in having optional empty?

Reply:
I actually used Optional to have a hands-on practice for me.  It is still a bit blurry for me when I should use optional and when not. If you want I can exclude the optional parts.